src/Raven.Server/Documents/Queries/AST/JavascriptCodeQueryVisitor.cs

```diff
@@ -95,9 +96,92 @@ public override void VisitCompoundWhereExpression(BinaryExpression @where)
 
         public override void VisitMethod(MethodExpression expr)
         {
+            // The regex usage here is just temporary, to get things working. Ideally, we want to implement the string functions in Jint
+            if (expr.Name.Value.Equals("startswith", StringComparison.OrdinalIgnoreCase) && expr.Arguments.Count == 2)
+            {
+                _sb.Append("new Regexp('^");
+                var param = expr.Arguments[1] as ValueExpression;
+                _sb.Append(Regex.Escape(param.Token.Value));
+                _sb.Append("').test('");
+                VisitExpression(expr.Arguments[0]);
+                
+                _sb.Append("')");
+                return;
+            }
+            
+            if (expr.Name.Value.Equals("endswith", StringComparison.OrdinalIgnoreCase) && expr.Arguments.Count == 2)
+            {
+                _sb.Append("new Regexp('");
+                var param = expr.Arguments[1] as ValueExpression;
+                _sb.Append(Regex.Escape(param.Token.Value));
+                _sb.Append("$').test('");
+                VisitExpression(expr.Arguments[0]);
+                
+                _sb.Append("')");
+                return;
+            }
+            
+            if (expr.Name.Value.Equals("intersect", StringComparison.OrdinalIgnoreCase) && expr.Arguments.Count >= 2)
```

Comment:
I don't understand how that is supposed to work

Reply:
actually, it works (except the <'> at the end, which caused an error). The query syntax also supports multiple parameters to "intersect"